### PR TITLE
Polish problem solving page functionality and interface

### PR DESCRIPTION
**Pull request by nicholashusin**
Changes included in this PR:
- Add test cases and alert usage in starter codes.
- Modify output sections in problem solving page to be scrollable to prevent overflow.
- Add highlighting functionality for running code.
- Add linting support to codemirror.

Screenshots:
- [Linter showing a warning symbol for missing semicolon (details are shown when mouse hovers over the symbol)](https://screenshot.googleplex.com/EJ1302f3MSP.png) 
- [Highlight showing section of the code that the interpreter is currently running](https://screenshot.googleplex.com/uNBo7jEdHAH.png)

### CODE REVIEW

**Review by lalaAdasia00 — APPROVED** [review]

File: scripts/solve.js [hunk 1]
```diff
@@ -175,3 +187,45 @@ async function submitSolution() {
...
+ **/
+function highlightCode(interpreter) {
+  let start = 0;
+  let end = 0;
+
+  if (interpreter.stateStack.length) {
+    let node    = interpreter.stateStack[interpreter.stateStack.length - 1].node;
+    start       = node.start;
+    end         = node.end;
+  }
```
Comment (ifeomi) [inline scripts/solve.js:223]: could you explain what this section of the code is doing? lgtm overall!

**Review by ifeomi — APPROVED** [review]

File: scripts/solve.js (hunk 1 above)
Comment (nicholashusin) [reply scripts/solve.js:223]: This is for getting what section of the code the interpreter is currently processing.
To be perfectly honest, I'm not perfectly sure what `stateStack` is about as documentation is scarce.

Looking at the way it works while executing the code, my understanding is that `stateStack` is the code [execution stack](https://blog.bitsrc.io/understanding-execution-context-and-execution-stack-in-javascript-1c9ea8642dd0). The `.node.start` and `.node.end` simply tells us what section of the code the execution stack is currently at.

Comment (chronologos) [reply scripts/solve.js:223]: good explanation; to add on, a `call stack` is a common concept in programming languages that spans from high-level programming languages to assembly.

a good SO [explanation](https://stackoverflow.com/questions/10057443/explain-the-concept-of-a-stack-frame-in-a-nutshell#:~:text=A%20stack%20frame%20is%20a,and%20space%20for%20local%20variables.)

**Review by chronologos — APPROVED** [review]

File: scripts/solve.js (hunk 1 above)
Comment (ifeomi) [reply scripts/solve.js:223]: ty both! @Nicholas2750 @chronologos

**Pull request merged**